pom.xml

```diff
@@ -91,6 +94,16 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-release-plugin</artifactId>
+                <version>3.0.0-M5</version>
+                <configuration>
+                    <allowTimestampedSnapshots>true</allowTimestampedSnapshots>
+                    <tagNameFormat>v@{project.version}</tagNameFormat>
+                    <pushChanges>false</pushChanges>
```

Comment:
```
<pushChanges>false</pushChanges>
```
Executing `mvn release:prepare --batch-mode` creates commits and tag on current branch. The line above  can be removed, so the commit is also automatically pushed without user interaction.  
As long as `pushChanges` is set to false, the user has to manually perform `git push && git push --tags`, so there is time to fix any potential errors without triggering release in github.